expr/func_grep.go

```diff
@@ -0,0 +1,49 @@
+package expr
+
+import (
+	"regexp"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+type FuncGrep struct {
+	in             GraphiteFunc
+	pattern        *regexp.Regexp
+	excludeMatches bool
+}
+
+func NewGrep() GraphiteFunc {
+	return &FuncGrep{excludeMatches: false}
+}
+
+func NewExclude() GraphiteFunc {
+	return &FuncGrep{excludeMatches: true}
+}
+
+func (s *FuncGrep) Signature() ([]Arg, []Arg) {
+	return []Arg{
+			ArgSeriesList{val: &s.in},
+			ArgRegex{key: "pattern", val: &s.pattern},
+		}, []Arg{
+			ArgSeriesList{},
+		}
+}
+
+func (s *FuncGrep) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncGrep) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	var outputs []models.Series
+	for _, serie := range series {
+		if s.pattern.MatchString(serie.Target) != s.excludeMatches {
+			outputs = append(outputs, serie)
+		}
+	}
+	return outputs, nil
```

Comment:
Is it okay to return an empty slice? Should we provide a warning if it's empty?

Reply:
Yes, an empty slice is fine. No need to warn.

Reply:
I looked around guess convention is to send back empty. (y)